Driver does not work with Linux 5.19.2

There are several compilation issues. Besides that, I tried the built in driver in the Linux kernel which still does not work. The connection is not established.

```sh
❯ make
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/5.19.2-arch1-1/build M=/home/matheus/rtl8821ce  modules
make[1]: Entering directory '/usr/lib/modules/5.19.2-arch1-1/build'
  CC [M]  /home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.o
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c: In function 'rtw_cfg80211_ch_switch_notify':
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:420:9: error: too few arguments to function 'cfg80211_ch_switch_notify'
  420 |         cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/matheus/rtl8821ce/include/osdep_service_linux.h:93,
                 from /home/matheus/rtl8821ce/include/osdep_service.h:50,
                 from /home/matheus/rtl8821ce/include/drv_types.h:27,
                 from /home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:17:
./include/net/cfg80211.h:7967:6: note: declared here
 7967 | void cfg80211_ch_switch_notify(struct net_device *dev,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c: In function 'cfg80211_rtw_scan':
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:2881:33: warning: the comparison will always evaluate as 'true' for the address of 'ssid' will never be NULL [-Waddress]
 2881 |                 if (ssids->ssid != NULL
      |                                 ^~
./include/net/cfg80211.h:2302:12: note: 'ssid' declared here
 2302 |         u8 ssid[IEEE80211_MAX_SSID_LEN];
      |            ^~~~
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c: At top level:
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:9593:20: error: initialization of 'int (*)(struct wiphy *, struct net_device *, unsigned int)' from incompatible pointer type 'int (*)(struct wiphy *, struct net_device *)' [-Werror=incompatible-pointer-types]
 9593 |         .stop_ap = cfg80211_rtw_stop_ap,
      |                    ^~~~~~~~~~~~~~~~~~~~
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:9593:20: note: (near initialization for 'rtw_cfg80211_ops.stop_ap')
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c: In function 'rtw_wdev_unregister':
/home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.c:9862:17: error: 'struct wireless_dev' has no member named 'current_bss'
 9862 |         if (wdev->current_bss) {
      |                 ^~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:249: /home/matheus/rtl8821ce/os_dep/linux/ioctl_cfg80211.o] Error 1
make[1]: *** [Makefile:1849: /home/matheus/rtl8821ce] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.19.2-arch1-1/build'
make: *** [Makefile:2244: modules] Error 2
```

--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -417,7 +417,11 @@ u8 rtw_cfg80211_ch_switch_notify(_adapter *adapter, u8 ch, u8 bw, u8 offset, u8 
 	if (ret != _SUCCESS)
 		goto exit;
 
+ #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
 	cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
+#else
+	cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0);
+#endif
 
 #else
 	int freq = rtw_ch2freq(ch);
@@ -4947,7 +4951,11 @@ static int cfg80211_rtw_change_beacon(struct wiphy *wiphy, struct net_device *nd
 	return ret;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
 static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev)
+#else
+static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev, unsigned int link_id)
+#endif
 {
 	_adapter *adapter = (_adapter *)rtw_netdev_priv(ndev);
 
@@ -9859,7 +9867,11 @@ void rtw_wdev_unregister(struct wireless_dev *wdev)
 	rtw_cfg80211_indicate_scan_done(adapter, _TRUE);
 
 	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)) || defined(COMPAT_KERNEL_RELEASE)
+	#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
 	if (wdev->current_bss) {
+	#else
+	if (wdev->connected) {
+	#endif
 		RTW_INFO(FUNC_ADPT_FMT" clear current_bss by cfg80211_disconnected\n", FUNC_ADPT_ARG(adapter));
 		rtw_cfg80211_indicate_disconnect(adapter, 0, 1);
 	}

